require 'cunn' fails on OSX 10.10

I'm on OS X 10.10

```
th> require 'nn'
true
                                                                      [0.0603s]
th> require 'cunn'
/usr/local/share/lua/5.1/trepl/init.lua:318: loop or previous error loading module 'cunn'
stack traceback:
	[C]: in function 'error'
	/usr/local/share/lua/5.1/trepl/init.lua:318: in function 'require'
	[string "require 'cunn'"]:1: in main chunk
	[C]: in function 'xpcall'
	/usr/local/share/lua/5.1/trepl/init.lua:587: in function 'repl'
	/usr/local/lib/luarocks/rocks/trepl/scm-1/bin/th:185: in main chunk
	[C]: at 0x010d8957b0
                                                                      [0.1432s]
th>
```

looking deeper, seems to be an issue with the libcunn.so file

```
th> require 'libtorch'
{
  ceil : function: 0x0dd82c58
  tan : function: 0x0dd82b90
  test : function: 0x0dd9ca38
  sinh : function: 0x0dd82b68
  loadobj : function: 0x0dd75408
... 

th> require 'libcunn'
/usr/local/share/lua/5.1/trepl/init.lua:318: /usr/local/share/lua/5.1/luarocks/loader.lua:117: error loading module 'libcunn' from file '/usr/local/lib/lua/5.1/libcunn.so':
	dlopen(/usr/local/lib/lua/5.1/libcunn.so, 6): no suitable image found.  Did find:
	/usr/local/lib/lua/5.1/libcunn.so: malformed mach-o image: load command #23 length (0) too small in /usr/local/lib/lua/5.1/libcunn.so
stack traceback:
	[C]: in function 'error'
	/usr/local/share/lua/5.1/trepl/init.lua:318: in function 'require'
	[string "require 'libcunn'"]:1: in main chunk
	[C]: in function 'xpcall'
	/usr/local/share/lua/5.1/trepl/init.lua:587: in function 'repl'
	/usr/local/lib/luarocks/rocks/trepl/scm-1/bin/th:185: in main chunk
	[C]: at 0x010d8957b0
```

I've run into an issue like this before with installing libraries like OpenCV, which required some .so surgery, but I'm a Lua newbie, so I'm not sure exactly how to proceed. OS X 10.10.1, Retina MBP 15".

Any suggestions or further information for debugging required would be appreciated! Would love to get to the bottom of this.

I'm not able to reinstall, I'm getting node.js errors

```
make: *** [Release/obj.target/fse/fsevents.o] Error 1
gyp ERR! build error
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/usr/local/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:267:23)
gyp ERR! stack     at ChildProcess.emit (events.js:98:17)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (child_process.js:820:12)
gyp ERR! System Darwin 14.0.0
gyp ERR! command "node" "/usr/local/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
gyp ERR! cwd /Users/Alex/torch-distro/extra/gfx.js/node_modules/fsevents
gyp ERR! node -v v0.10.35
gyp ERR! node-gyp -v v1.0.2
gyp ERR! not ok
```

Calling it a night, I'll get back at it sometime this week.
Are there clean uninstall steps?

--- a/install.sh
+++ b/install.sh
@@ -13,13 +13,31 @@ fi
 
 git submodule init
 git submodule update
+
+# NOTE: This gets most up-to-date packages. Desired behavior?
+# Submodules usually point to specific commits.
 git submodule foreach git pull origin master
+
+# If we're on OS X, first assert cmake is relatively recent,
+# and then delete the cmake directory for CUDA-dependent packages.
+# If this is not done on Yosemite,
+if [[ `uname` == "Darwin" ]]; then
+    # Check the dot version (currently only tested on Yosemite, 10.10)
+    osx_dotversion=$(sw_vers -productVersion | tr '.' "\n" | sed '2!d')
+    if [[ "$osx_dotversion" == "10" ]]; then
+        # This hurts me more than it hurts you
+        rm -rf ${currdir}/extra/cunn/cmake
+    fi
+fi
 
 mkdir -p build
 cd build
 cmake .. -DCMAKE_INSTALL_PREFIX="${PREFIX}" -DCMAKE_BUILD_TYPE=Release -DWITH_LUAJIT21=ON
 make && make install
 cd ..
+
+# Check for a CUDA install (using nvcc instead of nvidia-smi for cross-platform compatibility)
+path_to_nvcc=$(which nvcc)
 
 # check if we are on mac and fix RPATH for local install
 path_to_install_name_tool=$(which install_name_tool)
@@ -38,15 +56,12 @@ cd ${currdir}/pkg/paths && $PREFIX/bin/luarocks make rocks/paths-scm-1.rockspec
 cd ${currdir}/pkg/torch && $PREFIX/bin/luarocks make rocks/torch-scm-1.rockspec
 cd ${currdir}/pkg/dok && $PREFIX/bin/luarocks make rocks/dok-scm-1.rockspec
 cd ${currdir}/pkg/gnuplot && $PREFIX/bin/luarocks make rocks/gnuplot-scm-1.rockspec
-
 cd ${currdir}/exe/qtlua && $PREFIX/bin/luarocks make rocks/qtlua-scm-1.rockspec
 cd ${currdir}/exe/trepl && $PREFIX/bin/luarocks make
 cd ${currdir}/exe/env && $PREFIX/bin/luarocks make
-
 cd ${currdir}/extra/nn && $PREFIX/bin/luarocks make rocks/nn-scm-1.rockspec
 
-path_to_nvidiasmi=$(which nvidia-smi)
-if [ -x "$path_to_nvidiasmi" ]
+if [ -x "$path_to_nvcc" ]
 then
     cd ${currdir}/extra/cutorch && $PREFIX/bin/luarocks make rocks/cutorch-scm-1.rockspec
     cd ${currdir}/extra/cunn && $PREFIX/bin/luarocks make rocks/cunn-scm-1.rockspec
@@ -58,17 +73,11 @@ cd ${currdir}/pkg/sys && $PREFIX/bin/luarocks make sys-1.1-0.rockspec
 cd ${currdir}/pkg/xlua && $PREFIX/bin/luarocks make xlua-1.0-0.rockspec
 cd ${currdir}/pkg/image && $PREFIX/bin/luarocks make image-1.1.alpha-0.rockspec
 cd ${currdir}/pkg/optim && $PREFIX/bin/luarocks make optim-1.0.5-0.rockspec
-
 cd ${currdir}/extra/sdl2 && $PREFIX/bin/luarocks make rocks/sdl2-scm-1.rockspec
 cd ${currdir}/extra/threads && $PREFIX/bin/luarocks make rocks/threads-scm-1.rockspec
-
 cd ${currdir}/extra/graphicsmagick && $PREFIX/bin/luarocks make graphicsmagick-1.scm-0.rockspec
-
 cd ${currdir}/extra/argcheck && $PREFIX/bin/luarocks make rocks/argcheck-scm-1.rockspec
-
 cd ${currdir}/extra/audio && $PREFIX/bin/luarocks make audio-0.1-0.rockspec
-
 cd ${currdir}/extra/fftw3 && $PREFIX/bin/luarocks make rocks/fftw3-scm-1.rockspec
 cd ${currdir}/extra/signal && $PREFIX/bin/luarocks make rocks/signal-scm-1.rockspec
-
 cd ${currdir}/extra/nnx && $PREFIX/bin/luarocks make nnx-0.1-1.rockspec

--- a/install.sh
+++ b/install.sh
@@ -28,6 +28,11 @@ if [[ `uname` == "Darwin" ]]; then
         # This hurts me more than it hurts you
         rm -rf ${currdir}/extra/cunn/cmake
     fi
+
+    # Also, make sure that we build with Clang. CUDA's compiler nvcc
+    # does not play nice with any recent GCC version.
+    export CC=clang
+    export CXX=clang++
 fi
 
 mkdir -p build

